### ISSUE TITLE
Classification with an SVM model trained using SVM_light 

### ISSUE THREAD

**Issue by Afef00** [RyanMan1/PYNQ-SVM-OpenHW-2020#2]
Hello, 
I wanted to use this for a multiclass SVM (on MNIST dataset) on ZCU104 board . 
I trained the SVM model using SVM_light where I get a model file and a predictions file.
The model file contains the information regarding the training dataset like the size of the data used for training. It also contains information like the number of Support Vectors and their values, number of features, the bias value and the type of kernel used for training.
The predictions file contains all the values before applying the sign function of the decision function. These values are used for assigning the class which the test instance belongs to.
Is it possible to use your work using the given output files from svm_light?
Thanks!

**Comment by RyanMan1** [comment]
Hi there,

I would recommend having a look at the demo notebook and seeing if you can format your datasets into this format. Then the bitstream can be programmed and the SVM result calculated. Please let me know if you run into any issues.

Cheers!
Ryan

**Comment by Afef00** [comment]
Ok I will try that.
Thank you for the help!
Best regards

**Comment by Afef00** [comment]
Dear Ryan,
I have trained my SVM model in Python using scikit-learn (GridSearchCV). As I read in the documentation, the GridSearchCV uses one-versus-one SVM.
In Python, I could extract the following model parameters: alpha and bias values, number of support vectors (for each class). 
As I wanted to implement the prediction part (deployment) of SVM in HLS, I was trying use your source code in the HLS directory (test_predictions.cpp, test_predictions_TB.cpp, test_predictions.h).
I am trying to get the the files required (.dat files) to test the classifier and to convert them to the same  format as yours.  So I have some questions please:
1- In the ds_details.dat file,  it should include 3 values relating to the dataset: 
       // number of variables
   	// number of classes
 	// number of testing vectors
I don't get what do you mean by number of variables also considering number of testing vectors is it the number of samples used for the test ?
2- In the part **"read_test_predictions"** you used  test_predictions_libsvm.dat which as I understand contains predictions labels after you used LIBSVM. Is it necessary to use this part? I mean is it possible to only use the test labels?
3- Also I don't understand the use of geometric_values
4-My test data is with shape (number of samples, number of features)  is it possible to work with it?
Thanks!
